### ISSUE TITLE
Focus shifts to `leaflet-container` after activating markers when using a screen reader

### ISSUE THREAD

**Issue by Malvoz** [Leaflet/Leaflet#8114]
> **Warning**
>
> This issue is blocked on <s>[#8112](https://github.com/Leaflet/Leaflet/issues/8112) and</s> [#8115](https://github.com/Leaflet/Leaflet/issues/8115)</ins>, or there'll be accessibility regressions in some screen readers! This is because the side effect of the focus shift described here actually causes some screen readers to announce tooltips and popups.

- [x] I've looked at the [documentation](http://leafletjs.com/reference.html) to make sure the behavior is documented and expected
- [x] I'm sure this is a Leaflet code issue, not an issue with my own code nor with the framework I'm using (Cordova, Ionic, Angular, React…)
- [x] I've searched through the issues to make sure it's not yet reported

**Steps to reproduce**

1. Go to [quick-start/example.html](https://leafletjs.com/examples/quick-start/example.html)
2. Turn on a [screen reader of your choice](https://leafletjs.com/examples/accessibility/#test-your-maps) (note that you can pause speech by pressing <kbd>ctrl</kbd> at any time, if desired)
3. <kbd>Tab</kbd> to the marker and press <kbd>Enter</kbd>

**Expected behavior**

When activating the marker, focus should stay on the marker.

**Current behavior**

When activating the marker, focus is shifted to the map container.

This results in a reset of the focus order for keyboard users, and irrelevant and verbose announcements for screen reader users. It is a failure of WCAG success criterion [3.2.2 On Input](https://www.w3.org/TR/WCAG22/#on-input).

**Environment**

- Leaflet version: 1.7.1 / 1.8.0 / 1.9.4
- Browser (with version): Chrome 127.0.6533.90
- OS/Platform (with version): Windows 10
- Assistive tech: Narrator, NVDA

**Additional context**

  The current behavior has the side effect that popups actually are announced (although not _only_ the popup the user is interested in), so this issue is blocked on <s>https://github.com/Leaflet/Leaflet/issues/8112 and</s> https://github.com/Leaflet/Leaflet/issues/8115! Or there'll be accessibility regressions.

- The problem for end-users (who use screen readers) is pretty much the same as https://github.com/Leaflet/Leaflet/issues/7947.

**Comment by MichaelAllenWarner** [comment]
FYI, I don't see this behavior with VoiceOver on my Mac. Focus just remains on the Marker, as expected.
